echostream_managed_app/__init__.py

```diff
@@ -331,30 +327,35 @@ def __init__(
         self.__sdnotify = SystemdNotifier()
 
     async def __login(self, image_uris: list[str]) -> None:
-        registries = set[str]
+        registries = set()
         for image_uri in image_uris:
             registry = image_uri.split("/")[0]
             if registry != "public.ecr.aws":
                 registries.add(registry)
         registries = list(registries)
-        auth_tokens: list[str] = {
-            await _run_in_executor(self.ecr_public_client.get_authorization_token)[
-                "authorizationData"
-            ]["authorizationToken"],
-            *[
+        public_auth_token = (
+            await _run_in_executor(self.ecr_public_client.get_authorization_token)
+        )["authorizationData"]["authorizationToken"]
+        private_auth_tokens = (
+            [
                 auth_data["authorizationToken"]
-                for auth_data in await _run_in_executor(
-                    self.ecr_client.get_authorization_token,
-                    registryIds=[registry.split(".")[0] for registry in registries],
+                for auth_data in (
+                    await _run_in_executor(
+                        self.ecr_client.get_authorization_token,
+                        registryIds=[registry.split(".")[0] for registry in registries],
```

Comment:
Divided in to private and public, as private would be empty. and call boto3 call with empty registry id was giving error

Reply:
So only get auth token
 if registries
      else []